core/templates/domain/objects/UnitsObjectFactorySpec.ts

```diff
@@ -0,0 +1,184 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview unit tests for the units object factory service.
+ */
+
+import { ObjectsDomainConstants } from
+  'domain/objects/objects-domain.constants';
+import { Units, UnitsObjectFactory } from
+  'domain/objects/UnitsObjectFactory';
+
+describe('UnitsObjectFactory', () => {
+  describe('units object factory', () => {
+    var errors = null;
+    var units: UnitsObjectFactory = null;
+
+    beforeEach(() => {
+      units = new UnitsObjectFactory();
+    });
+
+    it('should test the CURRENCY_UNITS if the base unit is not null', () => {
+      ObjectsDomainConstants.CURRENCY_UNITS.dollar.base_unit = '100 cent';
```

Comment:
It's better to use `spyOn` instead of reassigning the value. Reassigning will create unexpected behavior, unless you set the original value on an `afterEach`, but it's better to always use `spyOn` since we use it in most of files.

Reply:
done!